dsa/arrays/Boyer-Moore Majority Voting Algorithm.md

```diff
@@ -0,0 +1,399 @@
+---
+id: Boyer-Moore-Majority-voting-algorithm
```

Comment:
replace `id: Boyer-Moore-Majority-voting-algorithm` to `id: boyer-moore-majority-voting-algorithm` for best practice